File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    cd = dict()
    for a, b in clothes:
        if b in cd:
            cd[b].append(a)
        else:
            cd[b] = [a]
    
    for a, b in cd.items():
        answer *= (1+len(b))
    return answer - 1
        

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        HashMap<String, Integer> clo = new HashMap<>();
        
        for(String[] c : clothes){
            clo.put(c[1], clo.getOrDefault(c[1], 0) + 1);
        }
        
        int answer = 1;
        for(String key : clo.keySet()){
            answer *= (clo.get(key) + 1);
        }
        return answer - 1;
    }
}